src/utils/utils_math.cpp

```diff
@@ -0,0 +1,24 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "utils_math.h"
+
+// Retrieves the position of the leftmost set bit.
```

Comment:
I think this can be replaced with `__builtin_clz`.

Reply:
As I remember correctly, TBB library has a similar approach. They implemented log2 for integers via this builtin if it is available.

Reply:
Done.